apps/cow-fi/services/cms/index.ts

```diff
@@ -146,6 +147,28 @@ export async function getArticles({
     throw error
   }
 
+  // If fetchAll is true and there are more pages, recursively fetch them
+  if (fetchAll && data.meta.pagination.page < data.meta.pagination.pageCount) {
```

Comment:
Just in case I would add a limit:
```
const ARTICLES_FETCHING_PAGE_LIMIT = 10
const { page, pageCount } = data.meta.pagination

if (fetchAll && page < pageCount && page < ARTICLES_FETCHING_PAGE_LIMIT)
```

100 * 10 = 1000 articles seems enough